javagen/src/main/java/com/azure/autorest/mapper/ModelPropertyMapper.java

```diff
@@ -87,6 +89,16 @@ public ClientModelProperty map(Property property) {
             }
         }
 
+        boolean isConstant = false;
+        String defaultValue = null;
+        if (property.getSchema() instanceof ConstantSchema) {
+            ConstantSchema constantSchema = (ConstantSchema) property.getSchema();
+            if (constantSchema.getValueType() instanceof StringSchema) {
+                isConstant = true;
```

Comment:
Shouldn't `isContant` be true even is the `valueType` is not of type `StringSchema`? `isConstant` should be set to true if `property.getSchema() instanceof ConstantSchema` is true.

Reply:
Yes, it should be. The defaultValue would also be different in this non-string case, and also that ConstantSchema would probably get a Class of its own.

The problem here is that currently I do not have any case that have a constant which is not a string, Hence I only do for string (mostly to unblock our team converting on management-plane SDK to v2).

That is reason I was asking 
> Do we have any swagger test on constant model property?

in https://github.com/Azure/autorest.java/issues/483

If you have any case (test, or real data-plane or management-plane swagger that have a non-string constant, please do let me know. Thanks.

Reply:
Well, I don't know of any swagger that has non-string constants but it should be fairly easy to create one. Take an existing swagger and replace one of the string constants with integer constants and update the schema for it. 

Also, you can go ahead and merge this change as this can be tested and updated in a separate PR for non-string constants.

Reply:
Got it. I will try to test non-string constant a bit later (after holiday), maybe try stealing some test case from Garrett's repo :-).

Currently still busying converting fluent java to azure core.